### ISSUE TITLE
Deleting users/groups after removal -- also how did you guys set uid/gid?

### ISSUE THREAD

**Issue by jpriddy** [SynoCommunity/spksrc#2495]
So I am at my wits end so I figured I would just ask in here directly.  I posted this to the r/synology subreddit as well:

https://www.reddit.com/r/synology/comments/586657/correct_way_to_adddelete_users_and_posibility_of/

Basically my issue boils down to 2 questions which arguably are more for Synology, but I figured you guys may have some expertise you could share:
1.  How did you guys forcibly set a uid and gid of these accounts on the synology for the various packages you ship?  I can not figure out what combination of options of the synouser command to use to achieve the same result.  Ideally I want to specify a user and proactively give it a UID, then also create a group with a specific GID then associate the user with that group and have it survive reboots and such without messing up any other users/groups on my synology.  I see no documented options with the synouser command to pass this info along.
2.  How can I delete this account (and others) that was created as part of the synocommunity package install:

root@nas:~# synouser --del sickbeard-custom
Lastest SynoErr=[user_db_delete.c:38]
synouser.c:745 SYNOLocalAccountUserDelete failed. synoerr=[0xB800]

That error points to:

ERR_RESERVED_USER   0xB800  The uid is less than UID_MIN.

according to this:  https://www.synology.com/en-global/support/developer#synouser

Naturally the developer guide doesn't really elaborate on how to forcibly remove and modify/set UID and GIDs...
### Setup

Package Name: Various -- sickbeard-custom, sabnzbd, etc 
Package Version: Unknown

NAS Model: 1812+
NAS Architecture: Atom
DSM version: DSM 6.0.2-8451 Update 2
### Expected behavior

Native tools should allow me to delete the user created as part of the package installs.
### Actual behavior

Obscure error that seemingly points to the uid/gids being outside the norm of what the synology tooling expects.
### Steps to reproduce
1. Remove synocommunity package that created a user
2. Try to manually delete user user synouser --del $USER

**Comment by BenjV** [comment]
Seen this document?
https://usdl.synology.com/download/Document/DeveloperGuide/DSM_Developer_Guide.pdf

**Comment by GoodOmens83** [comment]
Are you sure the user is showing up in /etc/passwd and /etc/shadow?

If you have one of the packages that install binaries such as deluser (e.g., ruTorrent among many others), you can just use that (`sudo ./deluser sickbeard-custom`).  The binaries are in the bin folder of the package (`/volume1/@appstore/rutorrent/bin`).

**Comment by Dr-Bean** [comment]
SynoCommunity packages do not use the `synouser` command. We use adduser/deluser (/addgroup/delgroup) provided via busybox. The reason for that is that the users that `synouser` creates are not fit to use with packages.
For DSM6, it will change: Synology has provided a method to create package users, which is described in the developers guide.
So, for question 1: we don't set uid/gid, and Synology doesn't provide a method for doing so. You can obviously just create a user, a group and add the one to the other, same as would happen via DSM.
For question 2: use `deluser` to delete SynoCommunity users. You're on your own after that, wrt support from us ;) For syntax, see any package that does user creation, e.g. https://github.com/SynoCommunity/spksrc/blob/master/spk/sickbeard-custom/src/installer.sh#L92

**Comment by Anorfwael** [comment]
Hello sorry to dig this one, but I'm stuck on an error I made : 
I added my user to synocommunity group with synogroup --memberadd synocommuunity my_user
Now, I can't remove it at all, no matter what.
The command deluser doesn't do anything, doesn't exist.

Anyone knows how to undo this ?
Thanks

**Comment by mreid-tt** [comment]
> Now, I can't remove it at all, no matter what. The command deluser doesn't do anything, doesn't exist.

I would caution against removing users. It may be useful to check if the user owns any files which may be affected by the user deletion with this command:
```
sudo find / -user my_user
```
Then, once you are sure you wish to delete the user it would be this command:
```
sudo /usr/syno/sbin/synouser --del my_user
```

**Comment by Anorfwael** [comment]
Actually it was to remove it from a group.
Found the solution after 2 hrs, and this was stupid...

I created a new user, and put him inside the group synocommunity. Then deleted the user.
Because when you use synogroup --memberadd group user  you in fact remove every other user from the group, to put only the one you typed in your command.

This was coded by people which brain is probably dead.
Thanks anyway
